### DISCUSSION TITLE
defineStore 的参数类型错误

### DISCUSSION THREAD

**Discussion by Capyhw** [vuejs/pinia#2724] [Help and Questions] [answered]
### Reproduction

https://play.pinia.vuejs.org/#eNqNU01v2zAM/SuCL0nRRGqQoQOMpMg29NAdtmHd0RfHZlKlNiVIdBog8H8fLcVu+oGiN4l8fHx8Eo/JN2vlvoEkTRa+cNqS8ECNFVWO22WWkM+Smwx1bY0jcRSNh3syDkQrNs7UYiRVYRokcJL8KMMMC4OeSQJoOeDHFxkuVOzAfHwhqG2VE/BNiMW6ITIoVkWli0fuG+olXl5y+zssHNSA3P8YiSWKtl2oWMQEC3XGlkySKHda51buvEEe7th1yU4JnikVIdLFrEadd5EsUeEswdfTtTNPnqfaMXjSQ1fslCphT8ZUfppbHcseiKxPlSpKZHwJld47iUAKba3e1Kyu5bWcqUqvFTdSGks4vGzDFdMS6s+w99DVlZx9kV8D635+Iq071o60zbBlW8jz42z09pUphamtrsD9tqT58V6Yk1eVefoZYuQaGCQWD1A8vhPf+UNU/ccB27eHs7Eod1ugmL69/wUHPg/J2pRNxegPkn/Bm6rpNEbY9wZLln2GC2rvwhtr3P7ztwcC9P1QndDgRsAHm398MPqz3Lmcn7n4/N/ZwmExSthofLUb4TfxUgwgB5shyd3DvsAhJOPaDOu1PCccj049R5MozxN/9VSML8TyphfseG0d9jchMBVX8cy6+6EJPA0j5qmY9XNlyAuatP8BsZ1auQ==

### Steps to reproduce the bug

正常引入pinia到vue项目中，然后编写store

### Expected behavior

我在编写pinia插件时，发现我无论使用setup还是option写法，在defineStore传递自定义参数时，预期是没有ts报错

### Actual behavior

这是官方文档的相关描述
https://pinia.vuejs.org/zh/core-concepts/plugins.html#adding-new-options
我在编写pinia插件时，发现我无论使用setup还是option写法，在defineStore传递自定义参数时，都会遇到ts报错
![image](https://github.com/user-attachments/assets/28b45af6-219a-4f7f-a56d-1fb278209448)
[https://play.pinia.vuejs.org/#eNqNU01v2zAM/SuCL0nRRGqQoQOMpMg29NAdtmHd0RfHZlKlNiVIdBog8H8fLcVu+oGiN4l8fHx8Eo/JN2vlvoEkTRa+cNqS8ECNFVWO22WWkM+Smwx1bY0jcRSNh3syDkQrNs7UYiRVYRokcJL8KMMMC4OeSQJoOeDHFxkuVOzAfHwhqG2VE/BNiMW6ITIoVkWli0fuG+olXl5y+zssHNSA3P8YiSWKtl2oWMQEC3XGlkySKHda51buvEEe7th1yU4JnikVIdLFrEadd5EsUeEswdfTtTNPnqfaMXjSQ1fslCphT8ZUfppbHcseiKxPlSpKZHwJld47iUAKba3e1Kyu5bWcqUqvFTdSGks4vGzDFdMS6s+w99DVlZx9kV8D635+Iq071o60zbBlW8jz42z09pUphamtrsD9tqT58V6Yk1eVefoZYuQaGCQWD1A8vhPf+UNU/ccB27eHs7Eod1ugmL69/wUHPg/J2pRNxegPkn/Bm6rpNEbY9wZLln2GC2rvwhtr3P7ztwcC9P1QndDgRsAHm398MPqz3Lmcn7n4/N/ZwmExSthofLUb4TfxUgwgB5shyd3DvsAhJOPaDOu1PCccj049R5MozxN/9VSML8TyphfseG0d9jchMBVX8cy6+6EJPA0j5qmY9XNlyAuatP8BsZ1auQ==](url)

![image](https://github.com/user-attachments/assets/04665c5f-0d08-451d-842a-ea80acadac86)
[https://play.pinia.vuejs.org/#eNqNU01v2zAM/SuCL0nRRGqQoQOCpMg29NAdtmHd0RfHZlKlFiVIdBog8H8fLSVu+oGiF0MmH997pMRD9s05uWsgm2XzUHrtSASgxom6wM0izyjk2U2O2jjrSRxEE+CerAfRirW3RgykKm2DBF5SGOSYY2kxMEkELXr88CLHuUoKzMc/BMbVBQH/CTFfNUQWxbKsdfnIurFe4uUly99h6cEAsv4hEUsUbTtXqYgJ5uqMLRtlye7YFE5ug0Vu7tCp5McE9zQTMdLFnEZddJE8U/EsIZjxytunwF1tGTw6QZc8KVXBjqytw7hwOpU9ELkwU6qskPEV1HrnJQIpdEa9qVley2s5UbVeKRZSGivYv5ThinEF5jPsJ+jySk6+yK+RdTc9kpqOtSNtc2x5LBT4ctZ682oopTVO1+B/O9J8eS+GU9S1ffoZY+Qb6C2WD1A+vhPfhn1y/ccDj28HZ21R4TdAKX17/wv2fO6TxlZNzegPkn8h2LrpPCbY9wYrtn2Gi27v4h1r3PwLt3sCDKemOqNxGhEfx/zjg9af7U7l9GyKz++dR9gvRgVrja92I74mXooe5GHdJ1k97gvsYzKtTb9ei3PC4eCoORiJ4YVY3CSTqQQZy7zDK96wLup5fT2yGO8IG469HnsiCNQ3KEQxE5N0ZmD8MEXW/gdUB1xk](url)

但是插件中从options里是可以正常取到值的
![image](https://github.com/user-attachments/assets/dc97004b-d342-4766-9145-1ca2ec1810fb)
但是我确实遇到了这个问题，我不知道是我那里有问题

**Comment by posva** [comment, marked as answer]
It seems that TS is reporting the error that way because of the overloads. I don't think it's something that can be improved, but I will be very happy to accept a PR if that turns out to be possible!
